sim/paladin/consecration.go

```diff
@@ -43,9 +98,71 @@ func (paladin *Paladin) registerConsecrationSpell(sim *core.Simulation) {
 				Cost: baseCost,
 				GCD:  core.GCDDefault,
 			},
+			CD: core.Cooldown{
+				Timer:    cdTimer,
+				Duration: ConsecrationCDTime,
+			},
 		},
 
 		ApplyEffects: core.ApplyEffectFuncDot(consecrationDot),
 	})
-	consecrationDot.Spell = paladin.Consecration
+	consecrationDot.Spell = paladin.ConsecrationRank4
+}
+
+const SpellIDConsecrationRank1 int32 = 26573
+
+var ConsecrationRank1ActionID = core.ActionID{SpellID: SpellIDConsecrationRank1}
+
+func (paladin *Paladin) registerConsecrationSpellRank1(sim *core.Simulation, cdTimer *core.Timer) {
+	baseCost := 120.0
+
+	consecrationDot := core.NewDot(core.Dot{
+		Aura: paladin.RegisterAura(core.Aura{
+			Label:    "ConsecrationRank1",
+			ActionID: ConsecrationRank1ActionID,
+		}),
+		NumberOfTicks: 8,
+		TickLength:    time.Second * 1,
+		TickEffects: core.TickFuncAOESnapshot(sim, core.SpellEffect{
+			DamageMultiplier: 1,
+			ThreatMultiplier: 1,
+			BaseDamage:       core.BaseDamageConfigMagicNoRoll(8, 0.119),
+			OutcomeApplier:   core.OutcomeFuncTick(),
+			IsPeriodic:       true,
+		}),
+	})
+
+	paladin.ConsecrationRank1 = paladin.RegisterSpell(core.SpellConfig{
+		ActionID:    ConsecrationRank1ActionID,
+		SpellSchool: core.SpellSchoolHoly,
+
+		ResourceType: stats.Mana,
+		BaseCost:     baseCost,
+
+		Cast: core.CastConfig{
+			DefaultCast: core.Cast{
+				Cost: baseCost,
+				GCD:  core.GCDDefault,
+			},
+			CD: core.Cooldown{
+				Timer:    cdTimer,
+				Duration: ConsecrationCDTime,
+			},
+		},
+
+		ApplyEffects: core.ApplyEffectFuncDot(consecrationDot),
+	})
+	consecrationDot.Spell = paladin.ConsecrationRank1
+}
+
+func (paladin *Paladin) CastConsecrationRank(sim *core.Simulation, target *core.Target, rank proto.RetributionPaladin_Rotation_ConsecrationRank) bool {
```

Comment:
Instead of checking a switch every time, you could just have the registerConsecrateSpell() function create+return only the relevant rank of Consecrate.

Reply:
This works for the ret sim, but maybe it would be useful for the prot paladin sim to have multiple consecrate ranks available? I'll make the recommended change and leave it to the prot folks to decide

Reply:
Code should be simplified and only switches once when registering the spell